util/network/cniprovider/cni_linux.go

```diff
@@ -0,0 +1,70 @@
+package cniprovider
+
+import (
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/moby/buildkit/util/network"
+	"github.com/pkg/errors"
+)
+
+func (ns *cniNS) sample() (*network.Sample, error) {
```

Comment:
Can we move the parsing components for all the files into a util package maybe?

Since the files all follow similar formats, it would be good to keep them all together.

Reply:
These are not similar actually. Eg. these are all single value files (and optimization with openat to read them quicker) while groups have key=value files(cpu.stat,memory.stat) or custom formats.